cluster/photon-controller/setup-prereq.sh

```diff
@@ -118,40 +134,80 @@ function create-vm-flavor {
     local flavor_name=$1
     local flavor_spec=$2
     local rc=0
+    local output
 
     $PHOTON flavor list | grep "\t${flavor_name}\t" > /dev/null 2>&1 || rc=$?
     if [ $rc -eq 0 ]; then
         echo "Flavor ${flavor_name} already made, skipping"
     else
         echo "Making VM flavor ${flavor_name}"
-        $PHOTON -n flavor create --name "${flavor_name}" --kind "vm" --cost "${flavor_spec}" > /dev/null 2>&1
+        rc=0
+        output=$($PHOTON -n flavor create --name "${flavor_name}" --kind "vm" --cost "${flavor_spec}" 2>&1) || rc=$?
+        if [ $rc -ne 0 ]; then
+            echo "ERROR: Could not create vm flavor \"${flavor_name}\", exiting"
+            echo "Output from flavor creation:"
+            echo ${output}
+            exit 1
+        fi
     fi
 }
 
 function create-disk-flavor {
     local rc=0
+    local output
 
     $PHOTON flavor list | grep "\t${PHOTON_DISK_FLAVOR}\t" > /dev/null 2>&1  || rc=$?
     if [ $rc -eq 0 ]; then
         echo "Flavor ${PHOTON_DISK_FLAVOR} already made, skipping"
     else
         echo "Making disk flavor ${PHOTON_DISK_FLAVOR}"
-        $PHOTON -n flavor create --name ${PHOTON_DISK_FLAVOR} --kind "ephemeral-disk" --cost "${SETUP_DISK_FLAVOR_SPEC}" > /dev/null 2>&1
+        rc=0
+        output=$($PHOTON -n flavor create --name ${PHOTON_DISK_FLAVOR} --kind "ephemeral-disk" --cost "${SETUP_DISK_FLAVOR_SPEC}" 2>&1) || rc=$?
+        if [ $rc -ne 0 ]; then
+            echo "ERROR: Could not create disk flavor \"${PHOTON_DISK_FLAVOR}\", exiting"
+            echo "Output from flavor creation:"
+            echo ${output}
+            exit 1
+        fi
     fi
 }
 
 function create-image {
     local rc=0
     local image_id=""
+    local num_images
+    local output
 
-    $PHOTON image list | grep "\tkube-image\t" > /dev/null 2>&1 || rc=$?
+    $PHOTON image list | grep "\t${PHOTON_IMAGE}\t" | grep ERROR > /dev/null 2>&1 || rc=$?
+    if [ $rc -eq 0 ]; then
+        echo "Warning: You have at least one ${PHOTON_IMAGE} image in the ERROR state. You may want to investigate."
+        echo "Images in the ERROR state will be ignored."
+    fi
+
+    rc=0
+    num_images=$($PHOTON image list | grep "\t${PHOTON_IMAGE}\t" | grep READY | wc -l)
+    if [ "$num_images" -gt 1 ]; then
+        echo "Warning: You have more than one good ${PHOTON_IMAGE} image. You may want to remove duplicates."
+    fi
+
+    $PHOTON image list | grep "\t${PHOTON_IMAGE}\t" | grep READY > /dev/null 2>&1 || rc=$?
     if [ $rc -eq 0 ]; then
         echo "Image ${PHOTON_VMDK} already uploaded, skipping"
     else
         echo "Uploading image ${PHOTON_VMDK}"
-        $PHOTON image create $PHOTON_VMDK -n ${PHOTON_IMAGE} -i EAGER > /dev/null 2>&1
+        rc=0
+        output=$($PHOTON image create $PHOTON_VMDK -n ${PHOTON_IMAGE} -i EAGER 2>&1) || rc=$?
```

Comment:
What about Pending delete? What if images of same name have one pending delete. I would recommend using image Id instead or save it as the result of this command for future usage. So, that VM creation doesn't fail due to any other images in the system.